android/app/src/main/java/org/smartregister/fhircore/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package org.smartregister.fhircore.viewmodel
+
+import android.accounts.Account
+import android.accounts.AccountManager
+import android.app.Application
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.View
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import com.google.android.material.internal.TextWatcherAdapter
+import kotlinx.android.synthetic.main.login.*
+import org.smartregister.fhircore.auth.account.AccountConfig
+import org.smartregister.fhircore.auth.account.AccountHelper
+import org.smartregister.fhircore.auth.account.OauthResponse
+import org.smartregister.fhircore.model.LoginUser
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+
+class LoginViewModel (application: Application) : AndroidViewModel(application), Callback<OauthResponse>{
+    var loginUser: LoginUser = LoginUser()
+    var allowLogin = MutableLiveData(false)
+
+    private fun shouldEnableLogin(): Boolean {
+        Log.i(javaClass.name, loginUser.username+" --- " + loginUser.password)
+
+        val username: String = loginUser.username
+        val password: String = loginUser.password
```

Comment:
String variables are immutable. For security _character arrays_ are recommended on all variables that store Passwords. Once the password is used the char array should then be overwritten e.g. using _asterisks_ to prevent a memory dump as an an attack vector.